platform/api/packet/api.go

```diff
@@ -249,6 +249,17 @@ func (a *API) CreateDevice(hostname string, conf *conf.Conf, console Console) (*
 		return nil, fmt.Errorf("timed out waiting for flatcar-install: %v", err)
 	}
 
+	if id != "" {
```

Comment:
For consistency and easier debugging if things go wrong depending on first use vs reuse, we could set AlwaysPXE from the beginning and rely on this code path here to unconditionally disable it again.

Reply:
(not a blocker)

Reply:
If the first PXE boot fails for what ever reason (ipxe or initramfs network error), on the second boot the instance would try to boot from disk. We can make this more robust if we start with `AlwaysPXE: &true` and disable it here again in all cases.